packages/dd-trace/src/appsec/blocking.js

```diff
@@ -100,29 +100,36 @@ function getBlockingData (req, specificType, actionParameters) {
 }
 
 function block (req, res, rootSpan, abortController, actionParameters = defaultBlockingActionParameters) {
-  if (res.headersSent) {
-    log.warn('[ASM] Cannot send blocking response when headers have already been sent')
-    return
-  }
+  try {
+    if (res.headersSent) {
+      log.warn('[ASM] Cannot send blocking response when headers have already been sent')
 
-  const { body, headers, statusCode } = getBlockingData(req, null, actionParameters)
+      throw new Error('Headers have already been sent')
+    }
 
-  rootSpan.addTags({
-    'appsec.blocked': 'true'
-  })
+    const { body, headers, statusCode } = getBlockingData(req, null, actionParameters)
 
-  for (const headerName of res.getHeaderNames()) {
-    res.removeHeader(headerName)
-  }
+    for (const headerName of res.getHeaderNames()) {
+      res.removeHeader(headerName)
+    }
 
-  res.writeHead(statusCode, headers)
+    res.writeHead(statusCode, headers)
 
-  // this is needed to call the original end method, since express-session replaces it
-  res.constructor.prototype.end.call(res, body)
+    // this is needed to call the original end method, since express-session replaces it
+    res.constructor.prototype.end.call(res, body)
 
-  responseBlockedSet.add(res)
+    rootSpan.setTag('appsec.blocked', 'true')
 
-  abortController?.abort()
+    responseBlockedSet.add(res)
+    abortController?.abort()
+
+    return true
+  } catch (err) {
+    rootSpan?.setTag('_dd.appsec.block.failed', 1)
```

Comment:
I think it's possible the 1 should be a string here, otherwise the tag goes into metrics instead of tags. Try to confirm it with other implementer of the spec